pkg/services/networkverifier/networkverifier.go

```diff
@@ -7,75 +7,17 @@ import (
 	"strings"
 	"time"
 
-	"github.com/aws/aws-sdk-go/aws/credentials"
 	v1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
-	hivev1 "github.com/openshift/hive/apis/hive/v1"
-
 	"github.com/openshift/configuration-anomaly-detection/pkg/aws"
-	"github.com/openshift/configuration-anomaly-detection/pkg/ocm"
 	"github.com/openshift/configuration-anomaly-detection/pkg/services/logging"
 
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
 	"github.com/openshift/osd-network-verifier/pkg/proxy"
 	"github.com/openshift/osd-network-verifier/pkg/verifier"
 	awsverifier "github.com/openshift/osd-network-verifier/pkg/verifier/aws"
 )
 
-// AwsClient is a wrapper around the aws client, and is used to import the received functions into the Provider
-type AwsClient = aws.Client
-
-// OcmClient is a wrapper around the ocm client, and is used to import the received functions into the Provider
-type OcmClient = ocm.Client
-
-// Provider should have all the functions that ChgmService is implementing
-type Provider struct {
-	// having awsClient and ocmClient this way
-	// allows for all the method receivers defined on them to be passed into the parent struct,
-	// thus making it more composable than just having each func redefined here
-	//
-	// a different solution is to have the structs have unique names to begin with, which makes the code
-	// aws.AwsClient feel a bit redundant\
-	*AwsClient
-	*OcmClient
-}
-
-// Service will wrap all the required commands the client needs to run its operations
-type Service interface {
-	// OCM
-	GetClusterInfo(identifier string) (*v1.Cluster, error)
-	GetClusterDeployment(clusterID string) (*hivev1.ClusterDeployment, error)
-	// AWS
-	GetSubnetID(infraID string) ([]string, error)
-	GetSecurityGroupID(infraID string) (string, error)
-	GetAWSCredentials() credentials.Value
-	IsSubnetPrivate(subnet string) bool
-}
-
-// Client refers to the networkverifier client
-type Client struct {
-	Service
-	Cluster           *v1.Cluster
-	ClusterDeployment *hivev1.ClusterDeployment
-}
-
-func (c *Client) populateStructWith(externalID string) error {
-	var err error
-
-	if c.Cluster == nil {
-		c.Cluster, err = c.GetClusterInfo(externalID)
-		if err != nil {
-			return fmt.Errorf("could not retrieve cluster info for %s: %w", externalID, err)
-		}
-	}
-
-	if c.ClusterDeployment == nil {
-		id := c.Cluster.ID()
-		c.ClusterDeployment, err = c.GetClusterDeployment(id)
-		if err != nil {
-			return fmt.Errorf("could not retrieve Cluster Deployment for %s: %w", id, err)
-		}
-	}
-	return nil
-}
+//go:generate mockgen --build_flags=--mod=readonly -source $GOFILE -destination ./mock/networkverifiermock.go -package networkverifiermock
```

Comment:
If we remove the mock mentioned in the comment above, please remove this as well.
Otherwise ignore :)